torch/_subclasses/functional_tensor.py

```diff
@@ -158,6 +161,15 @@ def __new__(cls, elem):
         )
         torch._C._set_throw_on_mutable_data_ptr(out)
         out.elem = elem
+
+        if torch.is_inference_mode_enabled():
+            if out.isRootTensor():
+                out._inference_mode_base = None
+                mode._storage_to_base[out.elem.storage()._cdata] = out
```

Comment:
you probably want to store a weakref.ref to the out.elem.storage(). It's possible that the tensor can go out of scope and the cdata can be reused.

Reply:
Or use a weakref.WeakKeyDictionary. The key should be the out.elem.storage()